src/__tests__/showInApp.test.ts

```diff
@@ -185,6 +186,114 @@ describe('defaults', () => {
   });
 });
 
+describe('Ask for web push notification permission', () => {
+  let notificationAPI: NotificationAPI;
+  let askForWebPushPermissionSpy: jest.SpyInstance<void, []>;
+
+  beforeEach(() => {
+    const settings = true;
+    Storage.prototype.getItem = jest.fn(() => JSON.stringify(settings));
+    server.connected;
+    const res: WS_EnvironmentDataResponse = {
+      route: 'environment/data',
+      payload: {
+        logo: 'string',
+        applicationServerKey: 'string',
+        askForWebPushPermission: true
+      }
+    };
+    server.send(res);
+    notificationapi.showInApp({
+      root: 'root'
+    });
+    askForWebPushPermissionSpy = jest.spyOn(
+      notificationapi,
+      'askForWebPushPermission'
+    );
+  });
+
+  afterEach(() => {
+    jest.restoreAllMocks();
+    if (notificationAPI) notificationAPI.destroy();
+  });
+  test('on Allow click', () => {
+    expect($('.notificationapi-opt-in-container')[0].style.display).toEqual('');
+    $('.notificationapi-allow-button').trigger('click');
+    expect($('.notificationapi-opt-in-container')[0].style.display).toEqual(
+      'none'
+    );
+    expect(askForWebPushPermissionSpy).toHaveBeenCalledWith();
+  });
+  test('on No thanks button click', () => {
+    expect($('.notificationapi-opt-in-container')[0].style.display).toEqual('');
+    $('.notificationapi-no-thanks-button').trigger('click');
+    expect($('.notificationapi-opt-in-container')[0].style.display).toEqual(
+      'none'
+    );
+  });
+});
+
+describe('Does not ask for web push notification permission when askForWebPushPermission is false ', () => {
+  let notificationAPI: NotificationAPI;
+  Storage.prototype.getItem = jest.fn(() => null);
+  beforeEach(() => {
+    server.connected;
+    const res: WS_EnvironmentDataResponse = {
+      route: 'environment/data',
+      payload: {
+        logo: 'string',
+        applicationServerKey: 'string',
+        askForWebPushPermission: false
+      }
+    };
+    server.send(res);
+    notificationapi.showInApp({
+      root: 'root'
+    });
+  });
+
+  afterEach(() => {
+    jest.restoreAllMocks();
```

Comment:
no mock to restore

Reply:
Storage.prototype.getItem = jest.fn(() => null);